pkg/cli/admin/ocpcertificates/regeneratemco/command.go

```diff
@@ -0,0 +1,128 @@
+
+	// ign* are for the user-data ignition fields
+	ignFieldIgnition = "ignition"
+	ignFieldSecurity = "security"
+	ignFieldTLS      = "tls"
+	ignFieldCA       = "certificateAuthorities"
+	ignFieldSource   = "source"
+	ignDataPrefix    = "data:text/plain;charset=utf-8;base64,"
+)
+
+var (
+	regenerateMCOLong = templates.LongDesc(`
+		Regenerate the Machine Config Operator certificates for an OCP v4 cluster.
+
+		More information about these certificates are in the product documentation;
+		visit [docs.openshift.com](https://docs.openshift.com)
+		and select the version of OpenShift you are using.
+
+		Experimental: This command is under active development and may change without notice.
+	`)
+
+	regenerateMCOExample = templates.Examples(`
+	    # Regenerate the MCO certs without modifying user-data secrets
+		oc adm certificates regenerate-mco --update-user-data=false
+
+		# Update the user-data secrets to use new MCS certs
+		oc adm certificates update-mco-user-data
+	`)
+)
+
+type RegenerateMCOOptions struct {
+	RESTClientGetter genericclioptions.RESTClientGetter
+
+	modifyUserData bool
+	outputNewCert  bool
+
+	genericclioptions.IOStreams
+}
+
+func NewCmdRegenerateTopLevel(restClientGetter genericclioptions.RESTClientGetter, streams genericclioptions.IOStreams) *cobra.Command {
+	o := &RegenerateMCOOptions{
+		RESTClientGetter: restClientGetter,
+		IOStreams:        streams,
+		modifyUserData:   true,
+	}
+
+	cmd := &cobra.Command{
+		Use:                   "regenerate-mco",
+		DisableFlagsInUseLine: true,
+		Short:                 i18n.T("Regenerate the machine config operator certificates in an OpenShift cluster"),
+		Long:                  regenerateMCOLong,
+		Example:               regenerateMCOExample,
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(o.Run(context.WithValue(context.Background(), certrotation.RunOnceContextKey, true)))
+		},
+	}
+
+	o.AddFlags(cmd)
+
+	return cmd
+}
+
+func NewCmdUpdateUserData(restClientGetter genericclioptions.RESTClientGetter, streams genericclioptions.IOStreams) *cobra.Command {
+	o := &RegenerateMCOOptions{
+		RESTClientGetter: restClientGetter,
+		IOStreams:        streams,
+		modifyUserData:   true,
+	}
+
+	cmd := &cobra.Command{
+		Use:                   "update-mco-user-data",
+		DisableFlagsInUseLine: true,
+		Short:                 i18n.T("Update user-data secrets in an OpenShift cluster to use updated MCO certfs"),
+		Long:                  regenerateMCOLong,
+		Example:               regenerateMCOExample,
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(o.RunUserDataUpdate(context.Background()))
+		},
+	}
+
+	return cmd
+}
+
+// AddFlags registers flags for a cli
+func (o *RegenerateMCOOptions) AddFlags(cmd *cobra.Command) {
+	cmd.Flags().BoolVar(&o.modifyUserData, "update-user-data", o.modifyUserData, "If true, automatically update user-data secrets in machine-api namespace. Not useful if node scaling not backed by MachineSet.")
```

Comment:
```suggestion
	cmd.Flags().BoolVar(&o.modifyUserData, "update-ignition", o.modifyUserData, "If true, automatically update user-data secrets in machine-api namespace. Not useful if node scaling not backed by MachineSet.")
```

Reply:
It's both...it's ignition in user data.  The actual API objects we are modifying are called e.g. `worker-user-data`.